[16364] Fullscreen view for work packages / fullscreen edit

furinvader opened the pull request:
The specification for this PR is stated in the description of the given work package under **Acceptance criteria: full edit**.
To be better suited for future feature requests while being less so for future and present bugs, the inplace-edit component is refactored.

https://community.openproject.org/work_packages/16364/activity

furinvader committed:
Implement work package editing in fullscreen view

furinvader committed:
Upgrade protractor to v2.4.0

furinvader committed:
Implement tests for WP fullscreen editing

furinvader committed:
Implement WP editing in list  overview

furinvader committed:
Add work package edit actions toolbar (save & cancel)

Saving the work package does not work as of yet.

furinvader committed:
Disable WP edit buttons for users without update permissions

furinvader committed:
Update lodash to 2.4.2

furinvader committed:
Refactor EditableFieldsService and EditPaneController

Move isActiveField and getPendingFormChanges to EditableFieldsState service.

furinvader committed:
Implement WP-edit save button

furinvader committed:
Make user input stay when switching between views

furinvader committed:
Reset user input when cancel button is used in edit mode

furinvader committed:
Change appearance of edit mode action buttons

furinvader committed:
Implement cancel button for fullscreen edit

Input values are set to their initial values when one restarts editing.

furinvader committed:
Change edit action toolbar appearance

furinvader committed:
Disable some ui elements when in edit mode

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-16364-fullscreen-edit.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://furinvader-16364-fullscreen-edit.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Ask user for confirmation when changing states

[ci skip]

furinvader committed:
Replace UI Select with default HTML elements

[ci skip]

furinvader committed:
Refactor inplace-edit directory structure

[ci skip]

furinvader committed:
Refactor component initiation of inplace-edit

Well, I actually moved a line in the HALAPIResource service,
too...

furinvader committed:
Reference field names via newly introduced field object

furinvader committed:
Reduce fieldController redundancies

[ci skip]

furinvader committed:
Keep inplace edit values between views

... while also keeping empty values

[ci skip]

furinvader committed:
Set fields to edit state when creating a new work pacakge

[ci skip]

furinvader committed:
Make inplaceEdit.from method an optional getter

Provide a length property to indicate the number of fields
associated with that form.
The changes also enhance testability.

[ci skip]
